src/compiler/parser.ts

```diff
@@ -594,7 +599,7 @@ namespace ts {
             // If this is a javascript file, proactively see if we can get JSDoc comments for
             // relevant nodes in the file.  We'll use these to provide typing informaion if they're
             // available.
-            if (isJavaScript(fileName)) {
+            if (isSourceFileJavaScript(sourceFile)) {
                 addJSDocComments();

```

Comment:
The problem there is that then you don't have these AST nodes until binding happens.  This means people observe a mutable tree that they need to know they need to get a binder for if they want to get to the JS doc nodes.